instrumentation-tools/nightly_tests_code_coverage.py

```diff
@@ -0,0 +1,95 @@
+#  Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+#  This product includes software developed at Datadog (https://www.datadoghq.com/).
+#  Copyright 2016-Present Datadog, Inc.
+
+import sys
+from argparse import ArgumentParser, Namespace, ArgumentTypeError
+from src.feature import Feature
+from src.api_surface import ApiSurface
+from src.constants import API_SURFACE_PATH
+from src.constants import NIGHTLY_TESTS_DIRECTORY_PATH
+from src.constants import NIGHTLY_TESTS_PACKAGE
+from src.constants import IGNORED_ENTITIES
+
+
+def restricted_float(x):
+    try:
+        x = float(x)
+    except ValueError:
+        raise ArgumentTypeError(f"{x} not a floating-point literal")
+
+    if x < 0.0 or x > 1.0:
+        raise ArgumentTypeError(f"{x} not in range [0.0, 1.0]")
+    return x
+
+
+def parse_arguments(args: list) -> Namespace:
+    parser = ArgumentParser()
+
+    parser.add_argument("-td", "--tests-directory-path",
+                        default=NIGHTLY_TESTS_DIRECTORY_PATH,
+                        help="The path to the nightly tests directory.")
+    parser.add_argument("-as", "--api-surface-paths",
+                        default=[API_SURFACE_PATH],
+                        nargs="+",
+                        help="The path to the api surface file.")
+    parser.add_argument("-t", "--test-coverage-threshold-to-assert",
+                        type=restricted_float,
+                        required=True,
+                        help="The test coverage threshold. It takes values from 0 to 1.")
+    parser.add_argument("-i", "--ignored-entities",
+                        nargs="+",
+                        default=IGNORED_ENTITIES,
+                        help="The entities to be ignored when traversing the apiSurface file.")
+
+    return parser.parse_args(args)
+
+
+def compute_test_coverage(
+        tests_directory_path: str,
+        api_surface_paths: [str],
+        threshold: int,
+        ignore_groups_reg_ex: str) -> int:
+    to_cover_test_cases = set()
+    api_surfaces = list(
+        map(lambda api_surface_path: ApiSurface(api_surface_path, ignore_groups_reg_ex), api_surface_paths))
+    for api_surface in api_surfaces:
+        to_cover_test_cases.update(api_surface.fetch_testable_methods())
```

Comment:
This bit seems slightly overengineered. Do you really need an `ApiSurface` class? Also you could simplify this by doing:

```python
    for api_surface in api_surfaces:
        to_cover_test_cases.update(fetch_testable_methods(api_surface, ignore_groups_reg_ex))
```

Reply:
it might be but then again I wanted to keep this a bit more modular until I see what else is needed here. I can always try to clean it at the end if needed.